subsys/bluetooth/host/hci_core.c

```diff
@@ -1271,6 +1269,53 @@ static int set_flow_control(void)
 	return bt_hci_cmd_send_sync(BT_HCI_OP_SET_CTL_TO_HOST_FLOW, buf, NULL);
 }
 #endif /* CONFIG_BT_HCI_ACL_FLOW_CONTROL */
+
+static int bt_clear_all_pairings(void)
+{
+	bt_conn_disconnect_all();
+
+	if (IS_ENABLED(CONFIG_BT_SMP)) {
+		bt_keys_clear_all();
+	}
+
+	if (IS_ENABLED(CONFIG_BT_BREDR)) {
+		bt_keys_link_key_clear_addr(NULL);
+	}
+
+	return 0;
+}
+
+int bt_unpair(const bt_addr_le_t *addr)
```

Comment:
Where exactly? Note that bt_unpair() is essentially the old bt_storage_clear() with the bt_storage->clear() called removed and moved to a new position in hci_core.c that's inside a `#if defined(CONFIG_BT_CONN)` section. Currently the API does not do any persistent clearing since there's no support for persistent pairing storage. As soon as it's added the API will do it, i.e. it'll then clear both runtime and persistent info, whereas currently it only clears runtime info (which is the only info we have for now).

Reply:
Of course, I failed to see that here. Disregard.